packages/bvaughn-architecture-demo/components/console/filters/FilterToggles.tsx

```diff
@@ -119,17 +121,14 @@ function Toggle({
   const id = `FilterToggle-${camelCase(label)}`;
   return (
     <div className={styles.Filter}>
-      <input
-        className={styles.Checkbox}
+      <Checkbox
+        data-test-id={id}
+        label={label}
+        secondaryLabel={count !== null && count > 0 && count}
```

Comment:
A `secondaryLabel` attribute feels very not-checkbox-like.

I think we should make the `label` prop accept a `ReactNode` (which includes strings) and then if we want to have a string + badge, that can just be passed in as the `label`.

Reply:
Can you expand more on why it doesn't feel checkbox-like?

The `label` prop currently accepts a `ReactNode`. This takes care of some the styling concerns around placing the secondary child. As always in API design I struggled a little with a name here. We could potentially use `addon` and `element` nomenclature similar to [ChakraUI](https://chakra-ui.com/docs/components/input/usage#left-and-right-addons) and other libraries like Bootstrap.

I'd like to tackle this in a follow-up though so it doesn't block anything if that's cool 😇

Reply:
Conceptually, checkboxes have attributes like: label, checked, change, disabled, etc.

The idea of "secondary label" is a thing that doesn't really have anything to do with a checkbox.

Just to be clear, all I'm asking is that we change:
```js
type CheckboxProps = {
  label: ReactNode;
  // ...
}
```

And then you can use it like:
```js
<Checkbox
  data-test-id={id}
  label={
    <>
      {label}
      {count !== null && <span className="...">{count}</span> > 0 && count}
  </>
  }
```

And we don't _leak_ in any non-standard concepts like "secondary label".

Reply:
Yeah, totally get the name can be confusing. I renamed `secondaryLabel` to `addon` for now, we could use `children`, but that feels a little weird to me. Going the `label` prop as the only route won't work, or at least will require managing layout each time which is what the `addon` prop handles.

Reply:
> Going the `label` prop as the only route won't work, or at least will require managing layout each time which is what the `addon` prop handles.

Could you explain?

My concern is that _the concept_ "primary" and "secondary" labels is _specific_ to the Console filters. It's not something a generic checkbox component should concern itself with.

To me, the main question is: What would you expect to happen if you clicked on the count (the "secondary label"?)
* If you'd expect nothing, then it shouldn't be passed to the checkbox at all.
* If you'd expect the checkbox to be toggled, then it should just be part of the "label".

Making the `label` prop a `ReactNode` (or using `children` – same thing really) leaves the flexibility of how to style the custom stuff (like the category counts) to the caller, and hopefully prevents us from adding more one-off props in the future.

In this case, it also gives us the flexibility to decide that the label count should truncate (with "...") but the count shouldn't. I think that's also kind of context specific.